[matthewken19] iP

matthewken19 opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://www.google.com/)).

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~SUPER FAST to use

All you need to do is,
1. download it from [here](https://www.google.com/).
2. double-click it
3. add your tasks
4. let it manage your tasks for you 👍 

And it is **FREE**!

Features:
- [x] Managing tasks
- [x] GUI
- [ ] Find deadlines (coming soon)
--- 
If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Added greeting and closing statements and line

Commit:
Added echo function

Commit:
Structuring of the line string, added list function, removed echo function

Commit:
Added helper Task class

Commit:
Added function to mark and unmark tasks

Commit:
added childclass Deadline, Event and Todo

Commit:
add case for todo, deadline, and event

Commit:
Changed local file location in runtest.bat

Commit:
Added input and expected.txt

Commit:
fixed the error: java file classpath should be in bin

Commit:
updated the expected.txt to include the greeting

Commit:
Added custom exception

Commit:
Added exception handling

Commit:
add delete function

Commit:
added the delete instruction format as part of error handling message

Commit:
indent left switch case to match the checkstyle

Commit:
added function to convert Task object to string that can be stored in text file

Commit:
added code for reading from text file, converting it to list of tasks and saving it back as a string in text file

Commit:
Merge branch 'branch-Level-7'

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        File dir;
```
```suggestion
        File directory;
```
Perhaps the name can be spelt out fully in case someone might not understand what "dir" stands for

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
-        System.out.println("Hello from\n" + logo);
+
+        File dir;
+        File f;
```
```suggestion
        File file;
```
Using "file" instead of "f" as the variable name would be preferable, especially since the variable is referenced elsewhere in the code. It can be challenging for people to recall the significance of "f" later on.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+
+        File dir;
+        File f;
+        ArrayList<Task> list = new ArrayList<>();
```
```suggestion
        ArrayList<Task> taskList = new ArrayList<>();
```
Can consider changing to a more meaningful name instead of "lists" as it can refer to any list instead of a specific list of tasks.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+        Scanner sc = new Scanner(System.in);
+        String input;
+        int index;
+        Task t;
```
```suggestion
        Task task;
```
Perhaps the name can be spelt out fully, same as the above.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+    }
+
+    private static ArrayList<Task> readFileContents(File filePath) throws FileNotFoundException {
+        ArrayList<Task> list = new ArrayList<>();
```
```suggestion
        ArrayList<Task> tasklist = new ArrayList<>();
```
Can consider changing to a more meaningful name instead of "lists" as it can refer to any list instead of a specific list of tasks.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+    private static ArrayList<Task> readFileContents(File filePath) throws FileNotFoundException {
+        ArrayList<Task> list = new ArrayList<>();
+        Scanner sc = new Scanner(filePath);
+        String s;
```
Perhaps can use a more meaningful name for this variable.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+                    stringSplit[i] = stringSplit[i].trim();
+                    System.out.println(stringSplit[i]);
+                }
+                boolean marked = Boolean.valueOf(stringSplit[1]);
```
```suggestion
                boolean isMark= Boolean.valueOf(stringSplit[1]);
```
Can consider changing the name of the boolean variable to sound like booleans.

AndrewOng2066 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,267 @@
+                    System.out.println(stringSplit[i]);
+                }
+                boolean marked = Boolean.valueOf(stringSplit[1]);
+                switch (stringSplit[0]){
```
```suggestion
                switch (stringSplit[0]) {
```
Missing a space before the curly bracket.

AndrewOng2066 reviewed (commented):
Overall, everthing is okay except some variable names might need to change.

Commit:
changed variable type to LocalDate instead of String and changed toString() method

Commit:
changed some naming style of variables

Commit:
Merge branch 'branch-Level-8'

Commit:
created Storage, TaskList, Ui class for more OOP

Commit:
added command classes for OOP

Commit:
added Storage, Parser, TaskList and Ui classes for OOP

Commit:
changed Duke.java using Storage, Parser, Ui, TaskList class for OOP

Commit:
Merge branch 'branch-MoreOOP'

Commit:
moved all java files into duke package

Commit:
Merge branch 'branch-Packages'

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
added java class for testing of Parser and TaskList

Commit:
changed main class location in build.gradle

Zack-Tay commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,58 @@
+
+    private String commandWord;
+    private String description;
+    static String defaultString = "Got it. I've added this task:";
```
I believe if this variable here is a constant it can be in all caps.

Zack-Tay commented on src/main/java/duke/AddCommand.java:
```diff
@@ -0,0 +1,58 @@
+
+    public void execute(TaskList tasks, Ui ui) throws DukeException {
+        Task task;
+        String[] stringSplit;
```
for naming convention sake maybe you want to consider changing it to something like stringComponents?

Zack-Tay commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.Scanner;
+
+public class Ui {
+    static String LINE = "_____________________________________________";
```
Good use of all CAPS for constants

Zack-Tay commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,65 @@
+package duke;
+
```
Might want to start adding javaDocs early to lesser workload in the coming weeks!

Zack-Tay commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+
+public class DukeException extends Exception {
```
If your bot name is Donald then ideally your Exception should be DonaldException rather than DukeException, same goes for the main class name!

Zack-Tay reviewed (commented):
Overall, I like the OOP abstraction! LGTM, but just take note of some changes that I have proposed that might help!

Commit:
added javadoc

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
added contains method in Task and find method in TaskList to support find command

Commit:
Merge branch 'branch-Level-9'

Commit:
edited the name of stringSplit to stringComponents to match coding standard

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Added Launcher and practice code for creating GUI

Commit:
result on duke.txt

Commit:
Practice on creating GUI, created MainWindow, Main, DialogBox java class and fxml files

Commit:
added GUI, input and output is as expected, but saving data is unhandled and GUI can improve

Commit:
added checkstyle xml files and edited corresponding lines in build.gradle

Commit:
Merge branch 'branch-CheckStyle'

Commit:
moved some files and organized packages

Commit:
Merge branch 'branch-GUI'

# Conflicts:
#	src/main/java/duke/Command.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/ui/Ui.java

Commit:
Parser and ParserTest class: extract DukeException for default switch case statement to Ui method

Long string message from DukeException should be handled by the Ui side to avoid cluttering.

Extracting the error message to the Ui side gives better OOP and abstraction.

Let's pull up default string messages output from other classes to Ui methods.

Using OOP and abstraction provides cleaner code for easier debugging in the future.

Commit:
Add JavaDoc to the intialization of Command and Task classes.

There are several command and task classes. The JavaDoc will provide information on the difference among the classes for better readability.

Commit:
Combine the two try catch clauses in load method into one.

The two try catch clauses catch similar exceptions (FileNotFoundException and IOException) and output the same code.

Combining the two clauses provide better readibility and reduce code duplication

Commit:
Add assertions

DeleteCommand and MarkCommand should have index > 0. Storage load method should also have file (create file if it does not exist).

Assertion of index > 0 ensures that the condition is always satisfied when assertion is turned on.
Assertion of file.exists() ensures that a file always exists after being created or read.

matthewken19 committed:
Merge pull request #1 from matthewken19/branch-A-CodeQuality

Branch code quality

matthewken19 committed:
Merge branch 'master' into branch-A-Assertions

matthewken19 committed:
Merge pull request #2 from matthewken19/branch-A-Assertions

Branch assertions

matthewken19 committed:
Add save method after processing the output of the input in Duke

matthewken19 committed:
Change the message to be more customised for the Ui

matthewken19 committed:
Merge branch 'branch-A-Personality'

matthewken19 committed:
Add background image jpg file

matthewken19 committed:
Change the Gui to be cat-themed and adjust the size of the window. JavaFX version is set to 21

matthewken19 committed:
Merge branch 'A-BetterGui'

matthewken19 committed:
Add equals method for Task classes and add checking of task duplication which raises an error

matthewken19 committed:
Move TaskList class to duke.task

matthewken19 committed:
Edit corresponding java classes that are affected from moving TaskList class to duke.task

matthewken19 committed:
Change duke project name to tofu

matthewken19 committed:
Add and modify JavaDoc to existing classes

matthewken19 committed:
Add more error handling for Storage class methods

matthewken19 committed:
Add Ui showcase image

matthewken19 committed:
Update README for Tofu the Cat program

matthewken19 committed:
Add icon for the program

matthewken19 committed:
Allow resizing of window

matthewken19 committed:
Add equals method for the Command classes and add more testing

matthewken19 committed:
Change java version to jdk 11
